packages/common/src/utils/fsHelperFunctions.ts

```diff
@@ -23,123 +23,137 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
 Ethereal Engine. All Rights Reserved.
 */
 
-import fs from 'fs'
+import { ObjectEncodingOptions, promises as fsp } from 'fs'
 import path from 'path'
-
-export function writeFileSyncRecursive(filename, content, charset = undefined) {
-  // -- normalize path separator to '/' instead of path.sep,
-  // -- as / works in node for Windows as well, and mixed \\ and / can appear in the path
-  let filepath = filename.replace(/\\/g, '/')
-
-  // -- preparation to allow absolute paths as well
-  let root = ''
-  if (filepath[0] === '/') {
-    root = '/'
-    filepath = filepath.slice(1)
-  } else if (filepath[1] === ':') {
-    root = filepath.slice(0, 3) // c:\
-    filepath = filepath.slice(3)
+import logger from '../../../server-core/src/ServerLogger'
```

Comment:
the common package must not import from server-core. there is a common logger that should be used instead.